homeassistant/components/flexmeasures/services.py

```diff
@@ -87,35 +89,46 @@ async def trigger_and_get_schedule(
         tzinfo = dt_util.get_time_zone(hass.config.time_zone)
         start = time_ceil(datetime.now(tz=tzinfo), resolution)
 
-        input_arguments = {
-            "sensor_id": get_from_option_or_config("power_sensor", entry),
-            "start": start,
-            "duration": get_from_option_or_config("schedule_duration", entry),
-            "soc_unit": "MWh",
-            "soc_min": get_from_option_or_config("soc_min", entry),
-            "soc_max": get_from_option_or_config("soc_max", entry),
-            "consumption_price_sensor": get_from_option_or_config(
+        flex_model = client.create_storage_flex_model(
+            soc_at_start=call.data.get("soc_at_start"),
+            soc_unit=SOC_UNIT,
+            soc_max=get_from_option_or_config("soc_max", entry),
+            soc_min=get_from_option_or_config("soc_min", entry),
+            soc_targets=call.data.get("soc_targets"),
+        )
+        flex_context = client.create_storage_flex_context(
+            consumption_price_sensor=get_from_option_or_config(
                 "consumption_price_sensor", entry
             ),
-            "production_price_sensor": get_from_option_or_config(
+            production_price_sensor=get_from_option_or_config(
                 "production_price_sensor", entry
             ),
-        }
-
-        LOGGER.info(input_arguments)
```

Comment:
This way good practice, to first create the kwargs dictionary, and then guarantee that the logger and the `client.trigger_and_get_schedule` get passed the exact same kwargs. It's also less verbose.